vm_supervisor/vm/firecracker/instance.py

```diff
@@ -103,10 +105,15 @@ async def setup(self):
             else [],
         )
 
-    async def wait_for_init(self) -> None:
+    async def wait_for_init(self) -> bool:
         """Wait for the init process of the instance to be ready."""
-        # TODO: Check availability via ping ?
-        return
+        if not self.vm_configuration:
+            raise ValueError("The VM has not been configured yet")
+
+        if not self.vm_configuration.ip:
+            raise ValueError("VM IP address not set")
+
+        return await ping(self.vm_configuration.ip, 3)
```

Comment:
This uses the IP assigned by firecracker internally, and I think that is better to use it to check if the VM is alive, because there will be edge cases where the IPv6 address doesn't work, but the VM is alive and running.